### PR TITLE
Fix shells and lmax argurments

### LINKED ISSUE #34: Shells and lmax flags
## The problem

Realized when testing out a different dataset that the `-shells` and `-lmax` flags aren't working as intended. (It tries to pass a python list instead of just a comma separated string). This bug prevents selection of individual shells to perform processing (if desired) and will need to be fixed in an upcoming version for this to work as intended.

As it currently stands, all of the shells of the respective datasets are used.

## Environment

* Operating System
* DBSC version

### PR DESCRIPTION
**Pull request by kaitj**
## Proposed changes

Resolves #34. 

Fixes the arguments to pass space-separated command line arguments (snakemake) as comma-separated to mrtrix rules (as necessary)

## Types of changes

What types of changes does your code introduce? Put an `x` in the boxes that apply

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionalitiy)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other (if none of the other choices apply)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you are unsure about any of the choices, don't hesitate to ask!

- [x] Changes have been tested to ensure that fix is effective or that a feature works.
- [ ] Changes pass the unit tests
- [ ] Code has been run through the `poe quality` task
- [ ] I have included necessary documentation or comments (as necessary)
- [ ] Any dependent changes have been merged and published

## Notes
All PRs will undergo the unit testing before being reviewed. You may be requested to explain or make additional changes before the PR is accepted.

### CODE REVIEW

**Comment by tkkuehn** [comment]
Do you mind merging from main so the linters work properly?

**Comment by kaitj** [comment]
> Do you mind merging from main so the linters work properly?

Rebased from main, workflows should work now.

File: dbsc/config/snakebids.yml [hunk 1]
```diff
@@ -128,7 +128,7 @@ parse_args:
   --lmax:
     help: '(Mrtrix3) the maximum spherical harmonic order for the output
           FOD(s). For algorithms with multiple outputs, this should be
-          provided as a comma-sperated list of integers, one for each output
+          provided as a space-sperated list of integers, one for each output
```
Comment (tkkuehn) [inline dbsc/config/snakebids.yml:131]: Unrelated to the changes at hand but while I'm looking at this, if `--lmax` can be ommitted, `nargs` should be `"*"`.

File: dbsc/workflow/rules/mrtpipelines.smk [hunk 2]
```diff
@@ -130,8 +130,8 @@ rule dwi2response:
         dwi=rules.nii2mif.output.dwi,
         mask=rules.nii2mif.output.mask,
     params:
-        shells=f"-shells {shells}" if shells else "",
-        lmax=f"-lmax {lmax}" if lmax else "",
+        shells=f"-shells {','.join(shells)}" if shells else "",
```
Comment (tkkuehn) [inline dbsc/workflow/rules/mrtpipelines.smk:133]: Loking at the help text (and `nargs`) of the `--shells` argument, it needs to be there, right? Maybe worth erroring out/raising an exception somewhere if this isn't defined.

**Review by tkkuehn — APPROVED** [review]
A couple of small suggestions relating to pre-existing issues, but the actual changes here look good to me.

File: dbsc/workflow/rules/mrtpipelines.smk (hunk 2 above)
Comment (kaitj) [reply dbsc/workflow/rules/mrtpipelines.smk:133]: Doesn't have to be required (similar to `lmax`).

**Pull request merged**